### PR TITLE
 Makes the lavaland mining base have more correct baseturfs

### LINKED ISSUE #34321: Mining base walls break to lava when deconstructed
Goliath punched wall, walls turned to lava.

### PR DESCRIPTION
**Pull request by ninjanomnom**
:cl: ninjanomnom
fix: Mining base walls when broken turn into plating which in turn breaks down to the basalt surface.
/:cl:

In order to make the walls and flooring have different baseturfs I had to make a baseturf helper variant that only changed the baseturfs of specific kinds of turfs in the area.

fixes #34321

### CODE REVIEW

File: code/modules/mapping/mapping_helpers.dm [hunk 1]
```diff
@@ -54,6 +56,35 @@
...
+
+/obj/effect/baseturf_helper/picky/Initialize()
+	if(!whitelist)
+		whitelist = list(loc.type)
+	else if(!islist(whitelist))
+		whitelist = list(whitelist)
+	else if(whitelist[whitelist[1]]) // Checking if it's a typecache-like list
+		return ..()
+	whitelist = typecacheof(whitelist)
+	return ..()
```
Comment (Cyberboss) [inline code/modules/mapping/mapping_helpers.dm]: . = ..() first in Initialize

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/modules/mapping/mapping_helpers.dm (hunk 1 above)
Comment (ninjanomnom) [reply code/modules/mapping/mapping_helpers.dm]: cant, this needs to be before the initialization

**Comment by F-OS** [comment]
does this fix shelters too?

**Comment by WJohn** [comment]
What about ruins? Do those turn into lava when destroyed?

**Comment by Kromgar** [comment]
@WJohn Yes they do. 

Shuttles, ruins and the like all have super fucked baseturfs. I saw a goliath punch out the ash walker ruin and lava tile was underneath.

The arrivals/escape shuttle is the worst though. Single weld and now there is a hole into space

**Comment by Cyberboss** [comment]
Hijacking your PR to test a webhook

**Comment by ninjanomnom** [comment]
Baseturf problems here are a mapping issue, this just introduces the tool to fix it and fixes the default lavaland map.

As for shelters they're probably using ChangeTurf instead of PlaceOnTop as would be proper, same goes for ruins but changing those is a different job.

**Comment by Iamgoofball** [comment]
uhh

cyberboss are you ok

**Comment by WJohn** [comment]
lol wtf happened

**Comment by ninjanomnom** [comment]
Cyber said why just before

**Pull request merged**